src/layer/AbstractSentinelHubV1OrV2Layer.ts

```diff
@@ -0,0 +1,128 @@
+  protected instanceId: string;
+  protected layerId: string;
+  protected evalscript: string | null;
+  protected evalscriptUrl: string | null;
+
+  public constructor(
+    instanceId: string,
+    layerId: string,
+    evalscript: string | null = null,
+    evalscriptUrl: string | null = null,
+    title: string | null = null,
+    description: string | null = null,
+  ) {
+    super(title, description);
+    if (!layerId || !instanceId) {
+      throw new Error('Parameters instanceId and layerId must be specified!');
+    }
+    this.instanceId = instanceId;
+    this.layerId = layerId;
+    this.evalscript = evalscript;
+    this.evalscriptUrl = evalscriptUrl;
+  }
+
+  protected getEvalsource(): string {
+    // some subclasses (Sentinel 1 at EO Cloud) might want to return a different
+    // evalsource depending on their parameters
+    return this.dataset.shWmsEvalsource;
+  }
+
+  public getMapUrl(params: GetMapParams, api: ApiType): string {
+    if (api !== ApiType.WMS) {
+      throw new Error('Only WMS is supported on this layer');
+    }
+    const baseUrl = `${this.dataset.shServiceHostname}v1/wms/${this.instanceId}`;
+    return wmsGetMapUrl(
+      baseUrl,
+      this.layerId,
+      params,
+      this.evalscript,
+      this.evalscriptUrl,
+      this.getEvalsource(),
+    );
+  }
+
+  protected getFindTilesAdditionalParameters(): Record<string, string> {
+    return {};
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected extractFindTilesMeta(tile: any): Record<string, any> {
+    return {};
+  }
+
+  public async findTiles(
+    bbox: BBox,
+    fromTime: Date,
+    toTime: Date,
+    maxCount: number = 50,
+    offset: number = 0,
+  ): Promise<PaginatedTiles> {
+    if (!this.dataset.searchIndexUrl) {
+      throw new Error('This dataset does not support searching for tiles');
+    }
+    const bboxPolygon = {
+      type: 'Polygon',
+      crs: { type: 'name', properties: { name: bbox.crs.urn } },
+      coordinates: [
+        [
+          [bbox.minY, bbox.maxX],
+          [bbox.maxY, bbox.maxX],
+          [bbox.maxY, bbox.minX],
+          [bbox.minY, bbox.minX],
+          [bbox.minY, bbox.maxX],
+        ],
+      ],
+    };
+    const payload = bboxPolygon;
+    const params = {
+      expand: 'true',
+      timefrom: fromTime.toISOString(),
+      timeto: toTime.toISOString(),
+      maxcount: maxCount,
+      offset: Number(offset),
+      ...this.getFindTilesAdditionalParameters(),
+    };
+
+    const url = `${this.dataset.searchIndexUrl}?${stringify(params, { sort: false })}`;
+    const response = await axios.post(url, payload, {
+      headers: { 'Content-Type': 'application/json' },
+    });
+
+    const responseTiles: any[] = response.data.tiles;
+    return {
+      tiles: responseTiles.map(tile => {
+        return {
+          geometry: tile.tileDrawRegionGeometry,
+          sensingTime: new Date(tile.sensingTime),
```

Comment:
This way should also be used in other `...Layer`s (`S2L2ALayer`, etc.), because now the `findTiles()` still returns the tiles with `sensingTime` as `string`.

Reply:
True, thanks, fixed.

I have also used ES6 return syntax to make the code a bit more succint.

Reply:
Nice, thank you. Resolving